Sources/Rendering/Core/CubeAxesActor/index.js

```diff
@@ -0,0 +1,831 @@
+import { glMatrix, vec3, mat4 } from 'gl-matrix';
+import * as d3 from 'd3-scale';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import macro from 'vtk.js/Sources/macro';
+import vtkActor from 'vtk.js/Sources/Rendering/Core/Actor';
+import vtkBoundingBox from 'vtk.js/Sources/Common/DataModel/BoundingBox';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import vtkMapper from 'vtk.js/Sources/Rendering/Core/Mapper';
+import vtkPixelSpaceCallbackMapper from 'vtk.js/Sources/Rendering/Core/PixelSpaceCallbackMapper';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkTexture from 'vtk.js/Sources/Rendering/Core/Texture';
+
+// ----------------------------------------------------------------------------
+// vtkCubeAxesActor
+// ----------------------------------------------------------------------------
+// faces are -x x -y y -z z
+// point 0 is 0,0,0 and then +x fastest changing, +y then +z
+const faceNormals = [
+  [-1, 0, 0],
+  [1, 0, 0],
+  [0, -1, 0],
+  [0, 1, 0],
+  [0, 0, -1],
+  [0, 0, 1],
+];
+const faceEdges = [
+  [8, 7, 11, 3],
+  [9, 1, 10, 5],
+  [4, 9, 0, 8],
+  [2, 11, 6, 10],
+  [0, 3, 2, 1],
+  [4, 5, 6, 7],
+];
+const edgePoints = [
+  [0, 1],
+  [1, 3],
+  [2, 3],
+  [0, 2],
+  [4, 5],
+  [5, 7],
+  [6, 7],
+  [4, 6],
+  [0, 4],
+  [1, 5],
+  [3, 7],
+  [2, 6],
+];
+const edgeAxes = [0, 1, 0, 1, 0, 1, 0, 1, 2, 2, 2, 2];
+const faceAxes = [
+  [1, 2],
+  [1, 2],
+  [0, 2],
+  [0, 2],
+  [0, 1],
+  [0, 1],
+];
+
+// some shared temp variables to reduce heap allocs
+const previousMatrixType = glMatrix.ARRAY_TYPE;
+glMatrix.setMatrixArrayType(Float64Array);
+const ptv3 = vec3.create();
+const pt2v3 = vec3.create();
+const tmpv3 = vec3.create();
+const tmp2v3 = vec3.create();
+const xDir = vec3.create();
+const yDir = vec3.create();
+const invmat = mat4.create();
```

Comment:
I see the `.create()` idea, but can't we just do `new Float64Array(3)` or  `new Float64Array(16)` instead?

Reply:
We can, but curious why would we want to? create() is basically their constructor.

Reply:
My take on that is that you don't know what you get depending on their implementation. (I know, you made the call to their API to change the default type). But since their API was meant to work on what ever array you have, it seems less opaque to state what we get natively rather than using an abstraction that hides what we get.

Reply:
OK I'll do it in a different MR and do it for all of vtk.js so we are consistent